### PR TITLE
Support mmseg:dev-1.x

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

Support mmseg:dev-1.x

## Modification

- [x] update task and End2EndModel in  `mmdeploy.codebase.mmseg.deploy`
- [x] update rewritings in  `mmdeploy.codebase.mmseg.models`
- [x] update tests for mmseg

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

### CODE REVIEW

File: mmdeploy/codebase/mmseg/deploy/segmentation_model.py [hunk 1]
```diff
@@ -82,78 +81,50 @@ def forward(self, img: Sequence[torch.Tensor],
...
+        batch_outputs = self.wrapper({self.input_name:
+                                      inputs})[self.output_names[0]]
+        return self.pack_result(batch_outputs, data_samples)
+
+    def pack_result(self, batch_outputs, data_samples):
+        predictions = []
+        for seg_pred, data_sample in zip(batch_outputs, data_samples):
+            # resize seg_pred to original image shape
+            if 'img_path' in data_sample.metainfo:
+                image = mmcv.imread(data_sample.metainfo['img_path'])
```
Comment (grimoire) [inline mmdeploy/codebase/mmseg/deploy/segmentation_model.py:99]: Do not use imread if you just want to get the shape of the image.
Use pillow instead.

File: mmdeploy/codebase/mmseg/deploy/segmentation_model.py [hunk 2]
```diff
@@ -167,42 +138,26 @@ def forward(self, img: Sequence[torch.Tensor],
...
-    return module.CLASSES, module.PALETTE
-
-
-def build_segmentation_model(model_files: Sequence[str],
-                             model_cfg: Union[str, mmengine.Config],
-                             deploy_cfg: Union[str, mmengine.Config],
-                             device: str, **kwargs):
+        if isinstance(inputs, list):
+            inputs = inputs[0]
+        if inputs.shape[2] != 3:
```
Comment (grimoire) [inline mmdeploy/codebase/mmseg/deploy/segmentation_model.py:143]: What's this for?

File: mmdeploy/pytorch/functions/adaptive_avg_pool.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+from torch.nn.modules.utils import _pair
+
+from mmdeploy.core import FUNCTION_REWRITER
+from mmdeploy.utils import (Backend, get_backend, get_root_logger,
+                            is_dynamic_shape)
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='torch.nn.functional.adaptive_avg_pool2d')
+def adaptive_avg_pool2d__default(ctx, input, output_size):
```
Comment (grimoire) [inline mmdeploy/pytorch/functions/adaptive_avg_pool.py:13]: Some backend do support adaptive_pooling like [ncnn](https://github.com/Tencent/ncnn/blob/master/docs/developer-guide/operators.md#pooling). @hanrui1sensetime

File: tests/test_codebase/test_mmseg/test_segmentation.py [hunk 4]
```diff
@@ -88,29 +78,32 @@ def test_build_backend_model(backend_model):
 
 
 def test_create_input():
-    inputs = task_processor.create_input(img, input_shape=img_shape)
+    img_path = 'tests/data/tiger.jpeg'
+    data_preprocessor = task_processor.build_data_preprocessor()
+    print(data_preprocessor, type(data_preprocessor))
```
Comment (grimoire) [inline tests/test_codebase/test_mmseg/test_segmentation.py:83]: remove the print

**Review by grimoire — APPROVED** [review]
LGTM

**Review by hanrui1sensetime — APPROVED** [review]
LGTM.

**Pull request merged**